packages/labs/ssr/src/lib/dom-shim.ts

```diff
@@ -160,5 +150,57 @@ export const getWindow = (
   window.window = window;
   window.global = window; // Required for node-fetch
 
+  if (includeJSBuiltIns) {
+    Object.assign(window, {
+      // No-op any async tasks
+      setTimeout() {},
+      clearTimeout() {},
+      // Required for node-fetch
+      Buffer,
+      console: {
+        log(...args: unknown[]) {
+          console.log(...args);
+        },
+        info(...args: unknown[]) {
+          console.info(...args);
+        },
+        warn(...args: unknown[]) {
+          console.warn(...args);
+        },
+        debug(...args: unknown[]) {
+          console.debug(...args);
+        },
+        error(...args: unknown[]) {
+          console.error(...args);
+        },
+        assert(bool: unknown, msg: string) {
+          if (!bool) {
+            throw new Error(msg);
+          }
+        },
+      },
+    });
+  }
+
   return window;
 };
+
+export const installWindowOnGlobal = (props: {[key: string]: unknown} = {}) => {
+  if (globalThis !== globalThis.window) {
```

Comment:
Add comment. Is this to make it idempotent?

Reply:
This was in there from when I was experimenting with react, or trying to get Polymer running in JSDOM, or something. It's basically to avoid installing a DOM shim if there's already one there, but don't exactly recall where it was needed. Will add a comment.

Reply:
Added comment